src/run_program.rs

```diff
@@ -1580,4 +1581,35 @@ mod tests {
 
         assert_eq!(result.unwrap().0, cost);
     }
+
+    #[test]
```

Comment:
Is it possible to move this to `collect_dialect`? It'd be nice to have no changes to `run_program.rs`

Reply:
Yeah I'll move it